### PR TITLE
[release/7.0] In-build signing for release branches

### PR DESCRIPTION
**Pull request by mmitche**

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -11,8 +11,12 @@ variables:
...
   - name: _CosmosToken
     value: C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==
-  - name: PostBuildSign
-    value: true
+  - ${{ if or(startswith(variables['Build.SourceBranch'], 'refs/heads/release/'), startswith(variables['Build.SourceBranch'], 'refs/heads/internal/release/')) }}:
+    - name: PostBuildSign
+      value: false
+  - ${{ else }}:
+    - name: PostBuildSign
+      value: true
```
Comment (dougbu) [inline azure-pipelines.yml:19]: Not sure I understand this part. Why not do in-build signing in at least manual internal builds of whatever branch❔ My preference is to make manual builds match official ones as much as possible.

Comment (mmitche) [reply azure-pipelines.yml:19]: @dougbu The intention here is to only run signing on branches that we could release. If a manual internal build requires signing they can always flip the switch.

Comment (mmitche) [reply azure-pipelines.yml:19]: @dougbu Any other comments?

Comment (dougbu) [reply azure-pipelines.yml:19]: nope

**Review by dougbu — APPROVED** [review]
@mmitche please use @dotnet/aspnet-build for infra reviews. I'm just part of the team.

**Pull request merged**